app/main.py

```diff
@@ -1,44 +1,63 @@
+import asyncio
 import time
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
-from iot.service import IOTService
+from iot.service import IOTService, run_sequence, run_parallel
 
 
-def main() -> None:
+async def main() -> None:
     # create an IOT service
     service = IOTService()
 
     # create and register a few devices
     hue_light = HueLightDevice()
     speaker = SmartSpeakerDevice()
     toilet = SmartToiletDevice()
-    hue_light_id = service.register_device(hue_light)
-    speaker_id = service.register_device(speaker)
-    toilet_id = service.register_device(toilet)
-
-    # create a few programs
-    wake_up_program = [
-        Message(hue_light_id, MessageType.SWITCH_ON),
-        Message(speaker_id, MessageType.SWITCH_ON),
-        Message(speaker_id, MessageType.PLAY_SONG, "Rick Astley - Never Gonna Give You Up"),
-    ]
 
-    sleep_program = [
-        Message(hue_light_id, MessageType.SWITCH_OFF),
-        Message(speaker_id, MessageType.SWITCH_OFF),
-        Message(toilet_id, MessageType.FLUSH),
-        Message(toilet_id, MessageType.CLEAN),
+    device = [
+        service.register_device(hue_light),
+        service.register_device(speaker),
+        service.register_device(toilet),
     ]
 
-    # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    hue_light_id, speaker_id, toilet_id = await asyncio.gather(*device)
+
+    await run_sequence(
+        run_parallel(
+            service.run_program(
+                [
+                    Message(hue_light_id, MessageType.SWITCH_ON),
+                    Message(speaker_id, MessageType.SWITCH_ON),
+                    Message(
+                        speaker_id,
+                        MessageType.PLAY_SONG,
+                        "Rick Astley - Never Gonna Give You Up"
+                    )
+                ]
+            )
+        )
+    )
+
+    await run_sequence(
+        run_parallel(
+            service.run_program(
+                [
+                    Message(hue_light_id, MessageType.SWITCH_OFF),
+                    Message(speaker_id, MessageType.SWITCH_OFF),
+                    Message(toilet_id, MessageType.FLUSH),
+                    Message(toilet_id, MessageType.CLEAN),
+                ]
+            )
+        )
+    )
```

Comment:
![image](https://github.com/mate-academy/py-asyncio-iot/assets/103201010/05c1bde9-28f0-4c93-b49f-d8907486a43a)
fix that